docs/pages/reference/join-methods.mdx

```diff
@@ -0,0 +1,427 @@
+---
+title: Join Methods and Token Reference
+description: Describes the different ways to configure a node to join the Teleport cluster.
+keywords: [join, joining, token]
+---
+
+<Admonition type="note" title="Prerequisite">
+You must be familiar with the [Teleport Core concepts](../core-concepts.mdx) before reading this page.
+</Admonition>
+
+## Definitions
+
+### Joining
+
+Joining a Teleport cluster is the act of establishing trust between a new node and
+all the existing nodes, already part of the Teleport cluster. This trust is
+established via certificates. Once a node has its certificates, it can interact
+with the other Teleport nodes.
+
+To request its certificates, a node must prove its identity to the Auth service.
+Teleport offers multiple ways for a joining node to prove its authenticity, they
+are called the Join Methods.
+
+### Join Methods
+
+A join method is a way for the Auth server to validate that a node requesting to
+join the Teleport cluster is legitimate. Some join methods are universal while
+others rely on the joining node context. For example cloud-provider
+join-methods (such as `iam`, `gcp` or `azure`) or CI-provider (such as `github`,
+`gitlab`, `cicleci`, `jenkins`) are more flexible and provide
+better security guarantees but require the joining node to run in a specific
+cloud-provider.
+
+Different join methods may provide different security guarantees. e.g. some
+join methods allow the joining node to request renewable certificates while other
+will require the node to join again to renew its certificate.
+
+The join method and its parameters are specified in the Token.
+
+### Token
+
+A Token is a Teleport resource that specifies which join method can be used in
+which context. For example, a token can allow SSH services to join with the
+`iam` join method if they are in the AWS account `333333333333` and can assume
+the role `teleport-node-role`:
+
+```yaml
+kind: token
+version: v2
+metadata:
+  name: my-iam-token
+spec:
+  roles: [Node]
+  join_method: iam
+  allow:
+  - aws_account: "333333333333"
+    aws_arn: "arn:aws:sts::333333333333:assumed-role/teleport-node-role/i-*"
+```
+
+<Admonition type="important">
+The token name may, or may not be sensitive depending on the join method.
+Secret-based join methods rely on the token name to be secret. In such
+cases the token name must be protected as knowing the token name in enough
+for a node to join the cluster.
+</Admonition>
+
+## Classification of join methods
```

Comment:
As written, this is a bit difficult to understand, as you basically create a 2x2 (4 total possibilities) by introducing static vs dynamic and then secret vs delegated.

I wonder if it would be simpler if we just didn't mention static at all. Then you only have to explain secret vs delegated. (We can put a small admonition that says "for backward compatibility Teleport also supports static tokens but their use should be avoided."

Reply:
I removed the static vs dynamic distinction